src/middlewares/errorHandler.ts

```diff
@@ -0,0 +1,13 @@
+import { NextFunction, Request, Response } from "express";
+
+export const errorHandler = (
+  err: Error,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  res.status ? res.status : res.status(500);
+  res.json({
+    message: err.message,
+  });
```

Comment:
I think this piece of code might show an error that cannot set headers when already sent. You could just create an error class with both the message and status code. Instead of the two lines in the image. And if the error is not an instance of the class you made then 500 server error.
 
<img width="407" alt="image" src="https://user-images.githubusercontent.com/112854405/230564640-c99d1d85-f8fe-4015-a31a-00af3b6fd585.png">